app/controllers/planning_applications/assessment/conditions_controller.rb

```diff
@@ -37,21 +43,36 @@ def update
       private
 
       def set_condition_set
-        @condition_set = @planning_application.condition_set
+        @condition_set = if @pre_commencement
+          @planning_application.pre_commencement_condition_set
+        else
+          @planning_application.condition_set
+        end
       end
 
       def set_conditions
         @conditions = @condition_set.conditions
       end
 
+      def set_pre_commencement
+        @pre_commencement = params[:pre_commencement] == "true"
+      end
```

Comment:
I think I would just make this a method rather than setting an instance variable. (And then no need to add a callback, etc.)

```suggestion
      def pre_commencement?
        params[:pre_commencement] == "true"
      end
```

Reply:
Not sure how that would work because I use the variable that gets set in the views?